### PR TITLE
Added more stuff to list meta queries

### PR DESCRIPTION
**Pull request by Vultraz**
With the goal of eventually using queries to populate the Admin UI, this adds more data that's currently passed through via `getAdminMeta` to the list metadata queries.

Adds:
- `description`
- `label`
- `singular`
- `plural`
- `path`
- ~`adminConfig`~
- A `mutation` object with the various list mutation names
- Changes the `queries` object to match the new `mutations` format (not entirely happy with these property names, might change)

Enables:
```gql
query {
  _UsersMeta {
    name
    label
    singular
    plural
    path
    schema {
      queries {
        item
        list
        meta
      }
      mutations {
        create
        createInput
        createMany
        createManyInput
        update
        updateInput
        updateMany
        updateManyInput
        delete
        deleteMany
      }
    }
  }
}
```

which gives

```js
{
  "data": {
    "_UsersMeta": {
      "name": "User",
      "label": "Users",
      "singular": "User",
      "plural": "Users",
      "path": "users",
      "schema": {
        "queries": {
          "item": "User",
          "list": "allUsers",
          "meta": "_allUsersMeta"
        },
        "mutations": {
          "create": "createUser",
          "createInput": "UserCreateInput",
          "createMany": "createUsers",
          "createManyInput": "UsersCreateInput",
          "update": "updateUser",
          "updateInput": "UserUpdateInput",
          "updateMany": "updateUsers",
          "updateManyInput": "UsersUpdateInput",
          "delete": "deleteUser",
          "deleteMany": "deleteUsers"
        }
      }
    }
  }
}
```

### CODE REVIEW

**Comment by Vultraz** [comment]
@timleslie finally got around to fixing the tests on this one (it's just the async timeout failure now).

**Comment by Vultraz** [comment]
Fixed conflicts.

**Comment by Vultraz** [comment]
Added `description`

**Comment by Vultraz** [comment]
Added `adminConfig` too. That should cover all the basic fields in `List.getAdminMeta`

File: packages/keystone/lib/ListTypes/list.js [hunk 1]
```diff
@@ -1483,18 +1489,33 @@ module.exports = class List {
...
-        const queries = [
-          this.gqlNames.itemQueryName,
-          this.gqlNames.listQueryName,
-          this.gqlNames.listQueryMetaName,
-        ];
+        const queries = {
+          item: this.gqlNames.itemQueryName,
+          list: this.gqlNames.listQueryName,
+          meta: this.gqlNames.listQueryMetaName,
+        };
```
Comment (MadeByMike) [inline packages/keystone/lib/ListTypes/list.js:1498]: What's the impact of changing the shape of this? I'm not up to speed on how the `getSchema ` method and `queries` are used. Also adding `mutations` I assume is just for consumption by developers? 

@timleslie assuming my reading is correct are you ok with exposing this API?

Comment (Vultraz) [reply packages/keystone/lib/ListTypes/list.js:1498]: `getSchema` helps populates the `schema` field in the `_ListMeta` type. I don't think `queries` is used for anything yet. Basically, there needs to be some way to access query/mutation names in the admin UI to dispatch the appropriate operations. Right now, the whole `gqlNames` object is passed through in the JSON global.

**Comment by MadeByMike** [comment]
@Vultraz I like it, let's keep exploring this in small steps.

**Comment by Vultraz** [comment]
Green! \o/

**Pull request merged**